x-pack/solutions/search/plugins/search_connectors/server/services/index.ts

```diff
@@ -234,20 +253,23 @@ export class AgentlessConnectorsInfraService {
   };
 }
 
-export const getConnectorsWithoutPolicies = (
+export const getConnectorsToDeploy = (
   packagePolicies: PackagePolicyMetadata[],
   connectors: ConnectorMetadata[]
 ): ConnectorMetadata[] => {
   return connectors.filter(
-    (x) => packagePolicies.filter((y) => y.connector_metadata.id === x.id).length === 0
+    (x) =>
+      packagePolicies.filter((y) => y.connector_settings.id === x.id).length === 0 &&
+      x.is_deleted === false
   );
 };
 
-export const getPoliciesWithoutConnectors = (
+export const getPoliciesToDelete = (
   packagePolicies: PackagePolicyMetadata[],
   connectors: ConnectorMetadata[]
 ): PackagePolicyMetadata[] => {
   return packagePolicies.filter(
-    (x) => connectors.filter((y) => y.id === x.connector_metadata.id).length === 0
+    (x) =>
+      connectors.filter((y) => y.id === x.connector_settings.id && y.is_deleted === true).length > 0
```

Comment:
nit: same, maybe let's split it up into separate boolean statements and join with `&&` in return

Reply:
instead of `filter` we can use `some` js lib function